### PR TITLE
[WIP] SNS: Updated logic to match state machine

### PR DESCRIPTION
**Pull request by mifrandir**
This is part of the changes required to make #37 work.

The changes so far are:
- (Em-)brakes: Add `kCruising` and merge `kNominalBraking` with `kEmergencyBraking`

Remaining questions:
- What is the [default behaviou](https://www.tutorialspoint.com/cplusplus/cpp_switch_statement.htm)r for these [switches](https://github.com/Hyp-ed/hyped-2021/blob/c8fc9a3c5fb1b70c32b86b00ff639a597641fa38/src/sensors/gpio_manager.cpp#L82-L116)? It's currently missing which leads to (a) compiler warnings and (b) potentially undesired behaviour. If no other cases can occur, we probably want to throw a critical failure if they do. If we want to skip other cases we could use some classic `default: break;`
- Most of the things related to STM in sensors seems to be part of the fake system. Unfortunately, I don't have a clue how this works and what we might need to change. @Videosquared and @hur maybe you can have a look at this? (Search in `src/sensors` for `kAccelerating`, then you'll see what I mean.) Also, try and give me an ELI5. :)

Maybe @robertasn can add to this as well? Afaik, telemetry requires the fake system to work to do tests so it may be in their interest to not brake it.

### CODE REVIEW

**Comment by hur** [comment]
The current changes look good to me. I'll let @Videosquared comment on the fake system since I've not worked with it.

**Review by hur — APPROVED** [review]

**Comment by robertasn** [comment]
I don't really have anything to add and I feel like not only Telemetry wants for everything to work, but the whole team should want for everything to work. But in this case I don't see anything wrong.

**Review by videosquared — APPROVED** [review]
The changes for this looks correct, I will try to compile something over the weekend (06-07/Mar) on trying to explain the fake system.
